src/crypto/BUILD.gn

```diff
@@ -104,6 +121,9 @@ static_library("crypto") {
   } else if (chip_crypto == "tinycrypt") {
     sources += [ "CHIPCryptoPALTinyCrypt.cpp" ]
     public_deps += [ "${mbedtls_root}:mbedtls" ]
+  } else if (chip_crypto == "platform") {
+    # Platform implementation is responsible for bringing their
+    # own implementation and dependencies
```

Comment:
This would be a circular dependency. It looks like now we have crypto depending on platform in a sense, but only by linkage. It's very possible now to build something where depending on the files touched, not all files are rebuilt properly, since the dependencies are missing here. They are "magically" done in `platform/FR32/BUILD.gn` but that is not actually setting the correct dependencies

Reply:
I've restructured the dependencies such that it is encoded appropriately now. The circular chain is broken by the fact that depending on `src/crypto` does not compile in a backend implementation when `chip_crypto == "platform"`.
All backend implementations now correctly depend on the public header and convenience functions from `src/crypto`, and `src/crypto` does not have a source dependency on the backend (if the backend source changes that does not impact the middleware). If somebody sets chip_crypto to platform and forgets to bring the backend implementation, the linker will tell.